src/python/deepgnn/migrate.py

```diff
@@ -0,0 +1,44 @@
+"""
```

Comment:
Create a new folder for this file, called `migrate` and rename file `to_0_1_56.py`. This way users can run upgrades across multiple versions.

Reply:
I think we should just leave the migrate.py file, if user installs pip package 0.1.56, the migrate script will pull them up to that version, if a user wants a migrate script for a different version, they can just pip install that version.

Reply:
How can you handle migration between any possible combination of version ranges? Are you going to put everything in one giant file and introduce extra arguments for each version?

Reply:
We can add a start-version parameter so they can input the initial version of the code. I think that will make it easier than running multiple files. This change it does not matter though, you can rerun the migrate script and it wont break.

Reply:
We can section the versions with comments. What is skip and override? If it fails half way through user will have to reset code and migrate again.

Reply:
Think about migration as stacked diffs/patches: split a big script into atomic changes and choose which ones you want to apply. Instead of a version control system, you have a python script to make transformations. If some changes are only relevant to e.g. python upgrades to async code, then an external user probably already applied and would rather skip them (simply delete relevant file). This structure is useful if there is a range of changes to apply and something must be skipped.

Reply:
I see what you are saying theoretically, but there should never be a real scenario where the user wants to skip changes correct? There will be scenarios where changes just dont get applied cause user never uses FeatureType or something. If we have one user looking for advanced usage they can copy the script and delete the relevant sections quickly. Maybe we can split the migrate script into multiple files when/if it is necessary?

Reply:
> there should never be a real scenario where the user wants to skip changes correct?

Hard to say, it is definitely not 0, because I am always surprised how code is used in the wild. I'd rather err on the side of caution, especially since all is needed from us is to move a file to a separate folder.